### ISSUE TITLE
motor_vehicle:conditional=private @ (xxxxx) not work doesn`t work as expected

### ISSUE THREAD

**Issue by xuruiray** [valhalla/valhalla#3967]
1, Private roads should have the lowest priority
I think we should set destination_only to true when hit conditional private 
https://github.com/valhalla/valhalla/blob/d6c80412c5fbfc8e85815a6081f68624b44c03e1/src/mjolnir/pbfgraphparser.cc#L1840-L1847

like this
```
       if (tmp == "private"){
          way_.set_destination_only(true);
       }
```

2, two scenario
for conditional private, I think there will be two scenario
```
motor_vehicle=no
motor_vehicle:conditional=private @ (16:30-07:00)
16:30-07:00 is a private road, other times are not allowed
```
We can support this function by set destination_only to true

```
motor_vehicle=yes
motor_vehicle:conditional=private @ (16:30-07:00)
16:30-07:00 is a private road, other times are allowed
```
for this scenario, I don't have any good ideas, can anyone provide an idea?

**Comment by nilsnolde** [comment]
`private` is mainly looked at in graph.lua. It's really easiest for us if you give the smallest routing example possible which shows the problem, ideally on openstreetmap.org, then we have smth to test with and provide recommendations easier.

**Comment by gknisely** [comment]
Few examples here:  https://overpass-turbo.eu/s/1ria

**Comment by xuruiray** [comment]
I use my own pbf file
![image](https://user-images.githubusercontent.com/15962843/218705975-c9bca8dd-156c-4622-bddc-68a5eeb01558.png)

motor_vehicle:conditional=private @ (16:30-07:00)

with curl in different time
```
curl http://localhost:9000/route --data '{"locations":[{"lat":38.03042735,"lon":-84.50985958, "radius":0},{"lat":38.03413174,"lon":-84.50859974, "radius":0}],"costing":"auto","format":"osrm","shape_format":"polyline5","directions_options":{"units":"miles"},"date_time":{"type":3,"value":"2023-01-24T06:00"}}'| jq
curl http://localhost:9000/route --data '{"locations":[{"lat":38.03042735,"lon":-84.50985958, "radius":0},{"lat":38.03413174,"lon":-84.50859974, "radius":0}],"costing":"auto","format":"osrm","shape_format":"polyline5","directions_options":{"units":"miles"},"date_time":{"type":3,"value":"2023-01-24T12:00"}}'| jq
```

the result is as follow
```
eybgFvyxbOy@UcD}@uAa@mA]uCy@oBi@uAa@
eybgFvyxbOy@UcD}@I`@ObAqA[mAWLkAFe@uCy@oBi@uAa@
```
![image](https://user-images.githubusercontent.com/15962843/218708282-3a6adfa3-b98f-476a-adc7-807bd91540de.png)
![image](https://user-images.githubusercontent.com/15962843/218708071-009cd846-fa47-4d7d-910c-69ab3691462e.png)

use this tool to show the route
https://developers.google.com/maps/documentation/utilities/polylineutility

I think this straight-line route is abnormal

This road is a private road when hit the conditions and the road should have the lowest priority and should not be taken in a straight line

When not hit the conditions, this road is not accessible，so we didn`t take it

------
ps: after add this code, this road are not be taken in a straight line
```
  if (tmp == "private"){
          way_.set_destination_only(true);
  }
```
i think there should have a new access type kTimedPrivate for this issue and do some change in cost file

**Comment by nilsnolde** [comment]
Yeah, I think some of the access tagging needs a bit more love in our code base. There's more missing things, like mode-dependent restrictions. Maybe we should even create a project for that. Once someone fixes _one_ thing wrt to some of that, it's often quite quick & easy to fix some others as well.

**Comment by xuruiray** [comment]
I think currently we only support time-conditional road access and turn restriction and i will try to resolve conditional private , and if it works I will submit a pr
eg:
{transportation mode}:conditional = {yes/no} @ (condition)
restriction:conditional = {turn_type} @ (condition)

**Comment by nilsnolde** [comment]
That'd be very appreciated, thanks!

**Comment by xuruiray** [comment]
https://github.com/valhalla/valhalla/pull/3983
I am not very professional with Valhalla, this is my initial idea, I hope you can make some comments， I will follow up and modify @nilsnolde
